[PyTorch] add test for numel slow path affecting data_ptr

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #100993

This test would have stopped #98090 -- data_ptr needs to call custom Python numel if it exists, since it could be arbitrary Python.

Differential Revision: [D45701566](https://our.internmc.facebook.com/intern/diff/D45701566/)

swolchok committed:
[PyTorch] add test for numel slow path affecting data_ptr

This test would have stopped #98090 -- data_ptr needs to call custom Python numel if it exists, since it could be arbitrary Python.

Differential Revision: [D45701566](https://our.internmc.facebook.com/intern/diff/D45701566/)

[ghstack-poisoned]

swolchok commented:
hm, I'm not sure why this test doesn't pass everywhere.

ezyang commented on test/test_python_dispatch.py:
```diff
@@ -2306,6 +2306,28 @@ def __torch_dispatch__(cls, func, types, args, kwargs):
 
+    def test_data_ptr_respects_numel_slow_path(self):
+        data = torch.randn(6, 2)
+        numel_called = [False]
```
this is dead

ezyang commented:
idk, it kind of looks like it is failing everywhere it is run

swolchok commented:
> idk, it kind of looks like it is failing everywhere it is run

weird, passed locally and on internal CI

swolchok commented:
@ezyang is there something internally enabling sym_numel hooking? I don't see any other tests that the functionality even works

ezyang commented:
If you're talking fbcode, this might be that macro where we disabled TensorImpl extensibility for perf. Would need an OSS build to trigger then.

swolchok commented:
maybe it needs a rebase to make sure we're atop an OSS commit that has the revert.

swolchok commented:
@pytorchbot rebase

swolchok committed:
Update on "[PyTorch] add test for numel slow path affecting data_ptr"


This test would have stopped #98090 -- data_ptr needs to call custom Python numel if it exists, since it could be arbitrary Python.

Differential Revision: [D45701566](https://our.internmc.facebook.com/intern/diff/D45701566/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] add test for numel slow path affecting data_ptr"


This test would have stopped #98090 -- data_ptr needs to call custom Python numel if it exists, since it could be arbitrary Python.

Differential Revision: [D45701566](https://our.internmc.facebook.com/intern/diff/D45701566/)

[ghstack-poisoned]

swolchok commented:
@pytorchbot merge
